src/Simplex/Chat/Library/Subscriber.hs

```diff
@@ -762,54 +762,68 @@ processAgentMessageConn vr user@User {userId} corrId agentConnId agentMessage =
+        -- possible improvement: check for each pending message, requires keeping track of connection state
+        unless (connDisabled conn) $ sendPendingGroupMessages user m conn
         withAgent $ \a -> toggleConnectionNtfs a (aConnId conn) $ chatHasNtfs chatSettings
         case memberCategory m of
           GCHostMember -> do
-            toView $ CRUserJoinedGroup user gInfo {membership = membership {memberStatus = status'}} m {memberStatus = status'}
-            let cd = CDGroupRcv gInfo m
+            (mStatus, membershipStatus) <- withStore' $ \db -> do
+              updateGroupMemberStatus db userId m GSMemConnected
+              membershipStatus <-
+                if not (memberPending membership)
+                  then updateGroupMemberStatus db userId membership GSMemConnected $> GSMemConnected
+                  else pure $ memberStatus membership
+              pure (GSMemConnected, membershipStatus)
+            toView $ CRUserJoinedGroup user gInfo {membership = membership {memberStatus = membershipStatus}} m {memberStatus = mStatus}
+            gcsi <- liftIO $ getNonMsgGCSI gInfo m
+            let cd = CDGroupRcv gInfo gcsi m
             createInternalChatItem user cd (CIRcvGroupE2EEInfo E2EInfo {pqEnabled = PQEncOff}) Nothing
             createGroupFeatureItems user cd CIRcvGroupFeature gInfo
-            let GroupInfo {groupProfile = GroupProfile {description}} = gInfo
-            memberConnectedChatItem gInfo m
-            unless expectHistory $ forM_ description $ groupDescriptionChatItem gInfo m
-            where
-              expectHistory = groupFeatureAllowed SGFHistory gInfo && m `supportsVersion` groupHistoryIncludeWelcomeVersion
+            memberConnectedChatItem gInfo gcsi m
+            unless (memberPending membership) $ maybeCreateGroupDescrLocal gInfo m
           GCInviteeMember -> do
-            memberConnectedChatItem gInfo m
-            toView $ CRJoinedGroupMember user gInfo m {memberStatus = status'}
+            mStatus <-
+              if not (memberPending m)
+                then withStore' $ \db -> updateGroupMemberStatus db userId m GSMemConnected $> GSMemConnected
+                else pure $ memberStatus m
+            gcsi <- liftIO $ getNonMsgGCSI gInfo m
+            memberConnectedChatItem gInfo gcsi m
+            toView $ CRJoinedGroupMember user gInfo m {memberStatus = mStatus}
             let Connection {viaUserContactLink} = conn
             when (isJust viaUserContactLink && isNothing (memberContactId m)) sendXGrpLinkMem
             when (connChatVersion < batchSend2Version) sendGroupAutoReply
-            unless (status' == GSMemPendingApproval) $ introduceToGroup vr user gInfo m
+            case mStatus of
+              GSMemPendingApproval -> pure ()
+              -- edge case: reviews were turned off mid connection;
+              -- options: proceed to review as declared, or introduce to group and send XGrpLinkAcpt;
```

Comment:
it is also more sensible - that review is disabled, does not need to auto-admit all members. We have multi-select, it could be a bulk action.